ls1intum/Artemis
# [Exam] Evaluate (quiz) exercises by instructor.

### Checklist

### Motivation and Context
Exams can contain quiz exercises. Normal quiz exercises are automatically evaluated at the quiz end using the `QuizScheduleService`.
As this service is not used, quizzes within exams are not evaluated automatically.
This PR adds a button `Evaluate quiz exercises` that allows the evaluation of all quiz exercises.

### Description
I've added a new `ExamQuizService` that allows for the evaluation of quiz exercises by loading the submissions from the database and storing the results again in the database.
The button `Evaluate quizzes` is displayed in the `Student exams` page and will only be enabled after the exam has ended
(**NOTE**: Currently, the exam end date is used, not the start date + max(workingTime)).
I've also disabled the `Generate individual exams` and `Prepare exercise start` buttons once the exam started.

The quiz statistics can then be viewed in the `Exercise groups` page using the `Statistics` button.

### Steps for Testing
1. Create an exam with one or multiple quiz exercises
2. Add students to the exam, generate the exams and prepare the exercises
3. Participate in the exam
4. After the exam is over, click on the `Evaluate quizzes` button on the `Student Exams` page.
5. Navigate to the `Exercise groups` page and review the statistics by clicking on `Statistics`. All quizzes should be evaluated.
6. Verify that the scores are calculated correct by clicking on `Scores` next to the exercise within the `Exercise groups` page.

### Screenshots
![image](https://user-images.githubusercontent.com/5084100/86339278-af71ce00-bc53-11ea-894e-1c3f9a6bd414.png)
![image](https://user-images.githubusercontent.com/5084100/86339380-cf08f680-bc53-11ea-8f65-1be47910413e.png)
